Fix all build.gradle files to use new deps declarations

foghina opened the pull request:
## Motivation (required)

We were getting a lot of deprecation warnings when building with gradle after the update to 4.1. This PR switches all `provided` to `compileOnly` and `compile` to `implementation` / `api`, depending on the use case.

## Test Plan (required)

CI + run showcase app.

oprisnik commented on imagepipeline-base-test/build.gradle:
```diff
@@ -36,7 +37,7 @@ task jarTest (type: Jar) {
 
 configurations {
-    testOutput.extendsFrom (testCompile)
+    testOutput.extendsFrom (testimplementation)
```
Don't you want `testImplementation`?

oprisnik commented on imagepipeline-base/build.gradle:
```diff
@@ -5,28 +5,30 @@ project.group = GROUP
-    testCompile("org.powermock:powermock-api-mockito:${POWERMOCK_VERSION}") {
+    implementation "com.parse.bolts:bolts-tasks:${BOLTS_ANDROID_VERSION}"
+    implementation project(':fbcore')
+    implementation 'com.facebook.soloader:soloader:0.1.0'
```
can you also do the same versioning as we do for the others? Also, should this be in this diff?

foghina commented:
Whoops, some of the changes from #1989 butted their way in here. Removed them and fixed the `testimplementation` thing. Not sure how the build was passing with that though.
